internal/services/network/virtual_network_resource.go

```diff
@@ -102,6 +103,26 @@ func resourceVirtualNetworkSchema() map[string]*pluginsdk.Schema {
 			},
 		},
 
+		"encryption": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
```

Comment:
we don't expose the `enabled` field, since this can be tracked by the presence of the block - can we remove this field (and make `unencrypted_allowed` required, since it'll be the only inner field)?

Reply:
updated